twitter4j-ads/src/twitter4jads/impl/TwitterAdsAudienceApiImpl.java

```diff
@@ -253,34 +244,50 @@ public boolean hasNext() {
             @Override
             public List<CustomAudienceOperation> next() {
                 List<CustomAudienceOperation> toReturn = Lists.newArrayList();
-                int totalSize = 0;
-                while (totalSize < CUSTOM_AUDIENCE_UPDATE_BATCH_SIZE && !operationPriorityQueue.isEmpty()) {
-                    int peekSize = operationPriorityQueue.peek().getUsers().size();
-                    if (totalSize + peekSize < CUSTOM_AUDIENCE_UPDATE_BATCH_SIZE) {
-                        totalSize += peekSize;
-                        toReturn.add(operationPriorityQueue.poll());
-                    } else {
-                        int diff = CUSTOM_AUDIENCE_UPDATE_BATCH_SIZE - totalSize;
-                        totalSize = CUSTOM_AUDIENCE_UPDATE_BATCH_SIZE;
-
-                        CustomAudienceOperation topOperation = operationPriorityQueue.poll();
-                        CustomAudienceOperation fractionalTopOperation = new CustomAudienceOperation();
-
-                        fractionalTopOperation.setOperationType(topOperation.getOperationType());
-                        fractionalTopOperation.setEffectiveFrom(topOperation.getEffectiveFrom());
-                        fractionalTopOperation.setExpireAt(topOperation.getExpireAt());
-
-                        Set<CustomAudienceUserDetails> fractionalTopOperationUsers = Sets.newHashSet();
-                        Iterator<CustomAudienceUserDetails> topOperationUsersIterator = topOperation.getUsers().iterator();
-                        while (diff > 0 && topOperationUsersIterator.hasNext()) {
-                            fractionalTopOperationUsers.add(topOperationUsersIterator.next());
-                            diff--;
+                long totalBytes = 0;
+                while (totalBytes < MAX_CUSTOM_AUDIENCE_PAYLOAD_BYTES && !operationPriorityQueue.isEmpty()) {
+                    totalBytes += CUSTOM_AUDIENCE_HEADER_BYTES;
+                    CustomAudienceOperation topOperation = operationPriorityQueue.poll();
+                    CustomAudienceOperation fractionalTopOperation = new CustomAudienceOperation();
+
+                    fractionalTopOperation.setOperationType(topOperation.getOperationType());
+                    fractionalTopOperation.setEffectiveFrom(topOperation.getEffectiveFrom());
+                    fractionalTopOperation.setExpireAt(topOperation.getExpireAt());
+
+                    Set<CustomAudienceUserDetails> fractionalTopOperationUsers = Sets.newHashSet();
+                    Iterator<CustomAudienceUserDetails> topOperationUsersIterator = topOperation.getUsers().iterator();
+
+                    while (topOperationUsersIterator.hasNext()) {
```

Comment:
I'll leave this up to you - this method seems pretty long. The while loop is a good opportunity to make it shorter but putting it in its own method.

Reply:
@symhom cleaned up the function a little by implementing `getSizeOfUserInBytes`. Would like to keep the while loops in the original function for simplicity.

Reply:
Looks good!